core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java

```diff
@@ -0,0 +1,910 @@
+    assertThat(updatedViewVersion.operation()).isEqualTo("replace");
+    assertThat(updatedViewVersion.representations()).hasSize(1).containsExactly(trino);
+    assertThat(updatedViewVersion.schemaId()).isEqualTo(OTHER_SCHEMA.schemaId());
+    assertThat(updatedViewVersion.defaultCatalog()).isEqualTo("default");
+    assertThat(updatedViewVersion.defaultNamespace()).isEqualTo(identifier.namespace());
+
+    assertThat(catalog().dropView(identifier)).isTrue();
+    assertThat(catalog().viewExists(identifier)).as("View should not exist").isFalse();
+  }
+
+  @Test
+  public void replaceViewVersionByUpdatingSqlForDialect() {
+    TableIdentifier identifier = TableIdentifier.of("ns", "view");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(identifier.namespace());
+    }
+
+    assertThat(catalog().viewExists(identifier)).as("View should not exist").isFalse();
+
+    SQLViewRepresentation spark =
+        ImmutableSQLViewRepresentation.builder()
+            .sql("select * from ns.tbl")
+            .dialect("spark")
+            .build();
+
+    View view =
+        catalog()
+            .buildView(identifier)
+            .withSchema(SCHEMA)
+            .withDefaultNamespace(identifier.namespace())
+            .withQuery(spark.dialect(), spark.sql())
+            .create();
+
+    assertThat(catalog().viewExists(identifier)).as("View should exist").isTrue();
+
+    ViewVersion viewVersion = view.currentVersion();
+    assertThat(viewVersion.representations()).hasSize(1).containsExactly(spark);
+
+    SQLViewRepresentation updatedSpark =
+        ImmutableSQLViewRepresentation.builder()
+            .sql("select * from ns.updated_tbl")
+            .dialect("spark")
+            .build();
+
+    // only update the SQL for spark
+    view.replaceVersion()
+        .withSchema(SCHEMA)
+        .withDefaultNamespace(identifier.namespace())
+        .withQuery(updatedSpark.dialect(), updatedSpark.sql())
+        .commit();
+
+    // history and view versions should reflect the changes
+    View updatedView = catalog().loadView(identifier);
+    assertThat(updatedView.history())
+        .hasSize(2)
+        .element(0)
+        .extracting(ViewHistoryEntry::versionId)
+        .isEqualTo(viewVersion.versionId());
+    assertThat(updatedView.history())
+        .element(1)
+        .extracting(ViewHistoryEntry::versionId)
+        .isEqualTo(updatedView.currentVersion().versionId());
+    assertThat(updatedView.versions())
+        .hasSize(2)
+        .containsExactly(viewVersion, updatedView.currentVersion());
+
+    // updated view should have the new SQL
+    assertThat(updatedView.currentVersion().representations())
+        .hasSize(1)
+        .containsExactly(updatedSpark);
+
+    assertThat(catalog().dropView(identifier)).isTrue();
+    assertThat(catalog().viewExists(identifier)).as("View should not exist").isFalse();
+  }
+
+  @Test
+  public void replaceViewVersionErrorCases() {
```

Comment:
I don't think there is a similar test case for errors when using the view builder. Can you add one?

Reply:
I've added a test for the create() and the replace() part of the view builder. In order to align error messaging with what we have in `ViewVersionReplace`, I've also slightly changed things up in the ViewBuilder implementation